File: 867/python/867.py
# python 867.py

class Solution(object):
    def transpose(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[List[int]]
        """
        tMat = []

        for i in range(len(matrix[0])):
            eMat = []
            for j in range(len(matrix)):
                eMat.append(matrix[j][i])
            tMat.append(eMat)

        return tMat

solution = Solution()

print(solution.transpose([[1, 2, 3], [4, 5, 6], [7, 8, 9]]))
print(solution.transpose([[1, 2, 3], [4, 5, 6]]))

File: 867/c#/867.cs
// mcs -out:867.exe 867.cs
// mono 867.exe

using System;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public int[][] Transpose(int[][] matrix)
    {
        List<List<int>> tMat = new List<List<int>>();
        List<int> eMat = new List<int>();

        for (int i = 0; i < matrix[0].Length; i ++) {
            eMat = new List<int>();
            for (int j = 0; j < matrix.Length; j ++) {
                eMat.Add(matrix[j][i]);
            }

            tMat.Add(eMat);
        }

        return tMat.Select(Enumerable.ToArray).ToArray();
    }

    static void Main(string[] args)
    {
        Solution solution = new Solution();
        Console
            .WriteLine("[{0}]",
            string
                .Join(", ",
                solution
                    .Transpose(new int[][] {new int[]{1,2,3},new int[]{4,5,6},new int[]{7,8,9}}).Cast<int[]>()));

        Console
            .WriteLine("[{0}]",
            string
                .Join(", ",
                solution
                    .Transpose(new int[][]{new int[]{1,2,3},new int[]{4,5,6}}).Cast<int[]>()));
    }
}
